### PR TITLE
Add new resource hint to all sdks for number of cpus per worker machine

### PR DESCRIPTION
**Pull request by kerrydc**
This resource hint allows for further specifcation of user needs at the transfor or pipeline level. 

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

File: sdks/python/.python-version [hunk 1]
```diff
@@ -0,0 +1 @@
+3.10.7
```
Comment (AnandInguva) [inline sdks/python/.python-version:1]: Remove this file.

File: sdks/python/apache_beam/transforms/resources.py [hunk 2]
```diff
@@ -177,6 +178,20 @@ def get_merged_value(
...
+  urn = resource_hints.CPU_COUNT.urn
+
+  @classmethod
+  def get_merged_value(
+      cls, outer_value, inner_value):  # type: (int, int) -> int
+    return ResourceHint._use_max(outer_value, inner_value)
+
+
+ResourceHint.register_resource_hint('cpu_count', CpuCountHint)
+ResourceHint.register_resource_hint('cpuCount', CpuCountHint)
```
Comment (AnandInguva) [inline sdks/python/apache_beam/transforms/resources.py:192]: add a comment `# Alias for interoperability with SDKs preferring camelCase.` on why 2 names for CpuCountHint for readability.

File: sdks/python/apache_beam/transforms/resources.py [hunk 3]
```diff
@@ -177,6 +178,20 @@ def get_merged_value(
 ResourceHint.register_resource_hint('minRam', MinRamHint)
 
 
+class CpuCountHint(ResourceHint):
+  """Describes desired hardware accelerators in execution environment."""
```
Comment (AnandInguva) [inline sdks/python/apache_beam/transforms/resources.py:182]:
```suggestion
  """Describes desired number of CPUs available in transform's execution environment."""
```

File: sdks/python/.python-version (hunk 1 above)
Comment (kerrydc) [reply sdks/python/.python-version]: Done.

File: sdks/python/apache_beam/transforms/resources.py (hunk 2 above)
Comment (kerrydc) [reply sdks/python/apache_beam/transforms/resources.py:193]: Done.

File: sdks/python/apache_beam/transforms/resources.py (hunk 3 above)
Comment (kerrydc) [reply sdks/python/apache_beam/transforms/resources.py]: Done.

**Review by AnandInguva — COMMENTED** [review]
LGTM!! Python changes look good to me

File: model/pipeline/src/main/proto/org/apache/beam/model/pipeline/v1/beam_runner_api.proto [hunk 4]
```diff
@@ -1982,5 +1982,9 @@ message StandardResourceHints {
     // SDKs should convert the size to bytes, but can allow users to specify human-friendly units (e.g. GiB).
     // Payload: ASCII encoded string of the base 10 representation of an integer number of bytes.
     MIN_RAM_BYTES = 1 [(beam_urn) = "beam:resources:min_ram_bytes:v1"];
+    // Describes desired number of CPUs available in transform's execution environment.
```
Comment (Abacn) [inline model/pipeline/src/main/proto/org/apache/beam/model/pipeline/v1/beam_runner_api.proto:1985]: would "CPU thread" be more accurate?

**Review by Abacn — COMMENTED** [review]
thanks, LGTM from java side. Just had a minor comment about documentation

File: model/pipeline/src/main/proto/org/apache/beam/model/pipeline/v1/beam_runner_api.proto (hunk 4 above)
Comment (kerrydc) [reply model/pipeline/src/main/proto/org/apache/beam/model/pipeline/v1/beam_runner_api.proto:1985]: This is the actual number of cpu or vcpu, threads is a different number.

**Comment by Abacn** [comment]
There is go test error:

```
# github.com/apache/beam/sdks/v2/go/pkg/beam/options/resource [github.com/apache/beam/sdks/v2/go/pkg/beam/options/resource.test]
Error: pkg/beam/options/resource/hint_test.go:167:30: undefined: pipepb.StandardResourceHints_CPU_COUNT
```

**Comment by lostluck** [comment]
LGTM for the Updated Go code, if we get the Go Tests precommit to pass.

**Comment by Abacn** [comment]
it appears  go test is failing on master

image

However the first a few red run was not go change.

**Comment by Abacn** [comment]
Run Java_PVR_Flink_Batch PreCommit

**Comment by Abacn** [comment]
Run Java PreCommit

**Comment by Abacn** [comment]
java precommit timeout due to #28957 not related - merging for now

**Pull request merged**